composer.lock

```diff
@@ -4079,8 +4079,7 @@
             },
             "require": {
                 "drupal/core": "~8.0",
-                "drupal/externalauth": "*",
-                "onelogin/php-saml": "~2.5"
```

Comment:
It's complicated, but there are reasons for this.
First of all, the way composer works (AFAIK) is that when we are building Drupal modules and their composer dependencies, they get in the common lock file at "some version".
drupal/build/web/modules/contrib/samlauth/composer.json
Adds this requirement:
  "require": {
    "onelogin/php-saml": "~2.5"
  }

This is actually a very old version which barely works with newer PHP. If the samlauth module is upgraded to 8-3.x-version, like it is on the courses page, we would get a newer and more compatible "onelogin/php-saml" version.

The problem becomes quite apparent in some environments, like my OSX local with PHP 7.3.8 which flat out refuses to run "composer update" when "onelogin/php-saml" is enabled and locked to "onelogin/php-saml": "~2.5".
It will complain because of lacking mcrypt PHP extension which has been deprecated in newer PHP (and is tbh too complicated to manually compile on my OSX Catalina with this PHP version).

It seems however that samlauth works ok with a somewhat newer version of "onelogin/php-saml:v2.13.0", and this can be required in the composer file with this:
  composer require onelogin/php-saml:v2.13.0 --ignore-platform-reqs

I still had to use "--ignore-platform-reqs", but mcrypt doesnt seem to be needed in our case as login works normally.

The intention is to get rid of samlauth and other stuff when refactoring login of guide, so for now I think we can keep circumventing annoying quirks of guide like this.